test/SlowTests/Issues/RavenDB_17513.cs

```diff
@@ -0,0 +1,128 @@
+        {
+            using var store = GetDocumentStore();
+
+            var requestExecutor = store.GetRequestExecutor();
+            using (var session = store.OpenSession())
+            {
+                session.Store(new Doc { Id = "doc-1" });
+                session.SaveChanges();
+            }
+
+            await LoadDataAsync(store);
+
+            var requests = requestExecutor.NumberOfServerRequests;
+
+            await LoadDataAsync(store); //got 'Not modified' from server - update httpCacheItem age to 0
+
+            Assert.Equal(requests, requestExecutor.NumberOfServerRequests);
+
+            await Task.Delay(3000); // cache timed out
+
+            await LoadDataAsync(store);
+
+            // additional request expected after cache timed out
+            Assert.Equal(requests + 1, requestExecutor.NumberOfServerRequests);
+
+            await LoadDataAsync(store);
+
+            // this should be aggressively cached without sending a request
+            Assert.Equal(requests + 1, requestExecutor.NumberOfServerRequests);
+        }
+
+        private async Task LoadDataAsync(IDocumentStore store)
+        {
+            using (var session = store.OpenAsyncSession())
+            using (session.Advanced.DocumentStore.AggressivelyCacheFor(TimeSpan.FromSeconds(2)))
+            {
+                var docLazy = session.Advanced.Lazily.LoadAsync<Doc>("doc-1");
+                var doc = await docLazy.Value;
+            }
+        }
+
+        [Fact]
+        public void AggressiveCacheWithTimeoutTest()
+        {
+            using var store = GetDocumentStore();
+
+            var requestExecutor = store.GetRequestExecutor();
+            using (var session = store.OpenSession())
+            {
+                session.Store(new Doc { Id = "doc-1" });
+                session.SaveChanges();
+            }
+
+            LoadData(store);
+
+            var requests = requestExecutor.NumberOfServerRequests;
+
+            LoadData(store); //got 'Not modified' from server - update httpCacheItem age to 0
+
+            Assert.Equal(requests, requestExecutor.NumberOfServerRequests);
+
+            Thread.Sleep(3000); // cache timed out
+
+            LoadData(store);
+
+            // additional request expected after cache timed out
+            Assert.Equal(requests + 1, requestExecutor.NumberOfServerRequests);
+
+            LoadData(store);
+
+            // this should be aggressively cached without sending a  request
+            Assert.Equal(requests + 1, requestExecutor.NumberOfServerRequests);
+        }
+
+        private void LoadData(IDocumentStore store)
+        {
+            using (var session = store.OpenSession())
+            using (session.Advanced.DocumentStore.AggressivelyCacheFor(TimeSpan.FromSeconds(2)))
+            {
+                var docLazy = session.Advanced.Lazily.Load<Doc>("doc-1");
+                var doc = docLazy.Value;
+            }
+        }
+
+        public class Doc
```

Comment:
let's try to keep nested classes private

Reply:
Done